### PR TITLE
Apply fixed random docker port per PCA to prevent change upon restart

### PR DESCRIPTION
**Pull request by amouhoub**
In this PR, we for docker to use a fixed random port number when forwarding ports to prevent assigning a new random port for PCA services upon restart (Pause -> Resume)
This is done programmatically by picking a free port randomly within the linux port random port range
Disclaimer: this works only on linux (the same is assumed for PCA anyway)

### CODE REVIEW

File: Docker/resources/catalog/Docker.xml [hunk 1]
```diff
@@ -84,6 +84,19 @@ variables.put("INSTANCE_NAME",variables.get("DOCKER_CONTAINER"))
             <![CDATA[
 echo BEGIN "$variables_PA_TASK_NAME"
 
+# Manually find a free random port to preserve it in case if docker (re)start (Pause/Resume)
+GET_RANDOM_PORT(){
+    read LOWERPORT UPPERPORT < /proc/sys/net/ipv4/ip_local_port_range
+    while :
+    do
+        RND_PORT="`shuf -i $LOWERPORT-$UPPERPORT -n 1`"
+        ss -lpn | grep -q ":$PORT " || break
```
Comment (tobwiens) [inline Docker/resources/catalog/Docker.xml]: PORT=9042

File: Cassandra/resources/catalog/Cassandra.xml [hunk 2]
```diff
@@ -55,6 +55,17 @@ PORT=9042
 ENV_VARS=$variables_ENV_VARS
 ################################################################################
 
+# Manually find a free random port to preserve it in case if docker (re)start (Pause/Resume)
+GET_RANDOM_PORT(){
+    read LOWERPORT UPPERPORT < /proc/sys/net/ipv4/ip_local_port_range
+    while :
+    do
+        RND_PORT="`shuf -i $LOWERPORT-$UPPERPORT -n 1`"
+        ss -lpn | grep -q ":$PORT " || break
```
Comment (tobwiens) [inline Cassandra/resources/catalog/Cassandra.xml]: PORT=9042

File: MySQL/resources/catalog/MySQL.xml [hunk 3]
```diff
@@ -52,6 +52,19 @@ $PASSWORD (Optional): Password for the root user. ]]>
             <![CDATA[
 echo BEGIN "$variables_PA_TASK_NAME"
 
+# Manually find a free random port to preserve it in case if docker (re)start (Pause/Resume)
+GET_RANDOM_PORT(){
```
Comment (tobwiens) [inline MySQL/resources/catalog/MySQL.xml]: Here the same

File: Cassandra/resources/catalog/Cassandra.xml [hunk 4]
```diff
@@ -55,6 +55,17 @@ PORT=9042
 ENV_VARS=$variables_ENV_VARS
 ################################################################################
 
+# Manually find a free random port to preserve it in case if docker (re)start (Pause/Resume)
+GET_RANDOM_PORT(){
```
[the remark this answers is not in the document]
Comment (amouhoub) [reply Cassandra/resources/catalog/Cassandra.xml]: @tobwiens change applied

File: MySQL/resources/catalog/MySQL.xml (hunk 3 above)
Comment (amouhoub) [reply MySQL/resources/catalog/MySQL.xml]: @tobwiens changes applied, thx !

**Pull request merged**